### ISSUE TITLE
MSB3030 Could Not Copy .dll from library

### ISSUE THREAD

**Issue by mlynch** [microsoft/WindowsAppSDK#2413]
### Describe the bug

When including a package in a Windows App SDK/Win UI 3 Class Library that includes a .DLL, and then installing that Class Library in a Windows App SDK/Win UI 3 C# App, the build fails to find and copy the DLL.

Thus, this chain does not work:

App Project -> Class Library -> 3rd party Nuget package -> DLL in 3rd party package

### Steps to reproduce the bug

1. Install the Windows App SDK including Visual Studio 2022 extension to add templates to VS
2. Create a new Class Library (WinUI 3 in Desktop) project. Install Couchbase.Lite.Enterprise into that library, pack it as a nupkg for distribution.
3. Create a new Blank App, Packaged (WinUI 3 in Desktop) project and install the library created in step 2.
4. Build the app and notice it fails with the above error.

### Expected behavior

The app builds correctly

### Screenshots

![cbl-net-2](https://user-images.githubusercontent.com/11214/163825285-a0f146ed-4142-422d-83a2-93714ce0276c.png)

### NuGet package version

1.0.0

### Packaging type

Packaged (MSIX)

### Windows version

Windows 10 version 21H2 (19044, November 2021 Update)

### IDE

Visual Studio 2022

### Additional context

If I look for the DLL in the library package, I see it here:

<img width="641" alt="cbl-net-4" src="https://user-images.githubusercontent.com/11214/163826051-96af0c50-c8b0-4b0e-8db7-b0766973a05b.png">

Windows App SDK version 1.0.2.

**Comment by borrrden** [comment]
I am interested in this as the original author of the package in question causing the issue (i.e. the "3rd party Nuget package" in question in the chain example).  There is likely some semantic changes that need to be accounted for but I haven't fully grasped what WinUI 3 means for native code shipped inside a C# nuget package.  Some background is that originally it started out for .NET Framework back in the 4.5 days, and so therefore the package had targets that would copy both the 32-bit and 64-bit x86 libraries into the build output folder, and then the C# library would choose which one to load based on which mode it was running in (a la `LoadLibraryEx` so that `DllImport` would use the one that we preloaded).

Then UWP came along and didn't like this, and instead compiled everything AOT and expected a single copy of the native library that matched the architecture being built.  So we made a UWP specific package that would include only the architecture being built alongside the assembly so that a simple `DllImport` was enough to load it.  

WinUI 3....I am unclear which of these approaches are going to be taken (or both?).  I'm also unclear why it is looking for our files inside of a different nuget package (or how they got there in the first place...).
